File: android/app/src/main/kotlin/com/example/floating_lyric/features/media_tracker/MediaState.kt
package com.example.floating_lyric.features.media_tracker

import android.os.Parcelable
import kotlinx.parcelize.Parcelize

@Parcelize
data class MediaState(
    var mediaPlayerName: String,
    var title: String,
    var artist: String,
    var album: String,
    var duration: Long,
    var position: Long,
    var isPlaying: Boolean,
) : Parcelable {

    fun toMap(): HashMap<String, Any> {
        val data: HashMap<String, Any> = HashMap()
        data["mediaPlayerName"] = mediaPlayerName
        data["title"] = title
        data["artist"] = artist
        data["album"] = album
        data["duration"] = duration.toDouble()
        data["position"] = position.toDouble()
        data["isPlaying"] = isPlaying

        return data
    }
}

File: lib/services/event_channels/media_states/media_state.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'media_state.freezed.dart';

@freezed
class MediaState with _$MediaState {
  const factory MediaState({
    required String mediaPlayerName,
    required String title,
    required String artist,
    required String album,
    required double position,
    required double duration,
    required bool isPlaying,
  }) = _MediaState;
}
